src/container/CardContainer.js

```diff
@@ -1,40 +1,184 @@
+     */
+    const prevOrder = curHeadCardOrder - cardNum;
+    const isLeakNFirstSlide = prevOrder >= -2 && prevOrder < 1;
+
+    const isFirstShowingSlide =
+      isLeakNFirstSlide || prevOrder === initialHeadCardOrder;
+
+    changeBtnActivation(
+      isFirstShowingSlide,
+      setDisabledPrevBtn,
+      setDisabledNextBtn
+    );
+
+    /**
+     * if(cardNum = 4) 1 2 3 | 4 5 6 7 | => 4 + 4 - (4 + 1) = 3
+     * if(cardNum = 3) 1 2 | 3 4 5 | => 3 + 3 - (3 + 1) = 2
+     */
+    const sizeToMove =
+      cardNum + curHeadCardOrder - (cardNum + initialHeadCardOrder);
+
+    setCurHeadCardOrder(
+      isLeakNFirstSlide
+        ? curHeadCardOrder - sizeToMove
+        : curHeadCardOrder - cardNum
+    );
+
+    const prevSlidingSize = getSlidingSize(
+      isLeakNFirstSlide,
+      sizeToMove,
+      cardNum
+    );
+
+    setSlidingSize(slidingSize + prevSlidingSize);
+  };
+
+  const handleClickNext = () => {
+    if (disabledNextBtn) return;
+
+    const cardInfosLen = cardInfos.length;
+    const showingHeadCardOrder = curHeadCardOrder + cardNum;
+    const sizeToMove = cardInfosLen % cardNum;
+
+    const isUndividedLastSlide =
+      (showingHeadCardOrder - initialHeadCardOrder) / cardNum ===
+      Math.floor(cardInfosLen / cardNum);
+
+    const isDividedLastSlide =
+      showingHeadCardOrder + cardNum === cardInfosLen + initialHeadCardOrder;
+
+    const isLastShowingSlide = isUndividedLastSlide || isDividedLastSlide;
+
+    changeBtnActivation(
+      isLastShowingSlide,
+      setDisabledNextBtn,
+      setDisabledPrevBtn
+    );
+
+    setCurHeadCardOrder(
+      isUndividedLastSlide
+        ? curHeadCardOrder + sizeToMove
+        : showingHeadCardOrder
+    );
+
+    const nextSlidingSize = getSlidingSize(
+      isUndividedLastSlide,
+      sizeToMove,
+      cardNum
+    );
+
+    setSlidingSize(slidingSize - nextSlidingSize);
+  };
 
-const CardContainer = ({ cardInfos, children, hasButton }) => {
   return (
     <StyledSection>
       {children}
-      <StyledCardContainer hasBtn={hasButton}>
-        {hasButton && <Button icon={'◀'} />}
-        {cardInfos.map((cardInfo, idx) => (
-          <StyledCard key={idx}>
-            <Card cardInfo={cardInfo} length={cardInfos.length} />
-          </StyledCard>
-        ))}
-        {hasButton && <Button icon={'▶'} />}
-      </StyledCardContainer>
+      {hasButton && (
+        <StyledLeftButton
+          icon={'◀'}
+          disabled={disabledPrevBtn}
+          onClick={handleClickPrev}
+        />
+      )}
+      {hasButton && (
+        <StyledRightButton
+          icon={'▶'}
+          disabled={disabledNextBtn}
+          onClick={handleClickNext}
+        />
+      )}
+      <StyledDiv>
+        <CardBox
+          hasBtn={hasButton}
+          cardNum={cardNum}
+          slidingSize={slidingSize}
+          cardInfos={cardInfos}
+        />
+      </StyledDiv>
     </StyledSection>
   );
 };
 
+const StyledDiv = styled.div`
+  margin: 0 40px 0 40px;
```

Comment:
```suggestion
  margin: 0 40px;
```
위아래 좌우는 보통 줄여서 많이 씁니다.